src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -507,6 +518,10 @@ private boolean isWrongOrder(String name, boolean isStatic) {
         if (isStatic && useContainerOrderingForStatic) {
             result = compareContainerOrder(lastImport, name, caseSensitive) > 0;
         }
+        else if (onDemandImportLastInGroup && (name.lastIndexOf(WILDCARD_GROUP_NAME) != -1
+                || lastImport.lastIndexOf(WILDCARD_GROUP_NAME) != -1)) {
+            result = compareOnDemandImport(lastImport, name, caseSensitive) > 0;
+        }
```

Comment:
move `name.lastIndexOf(WILDCARD_GROUP_NAME) != -1  || lastImport.lastIndexOf(WILDCARD_GROUP_NAME) != -1)` to static function and name it to help understand what it is doing by name.